test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ScalaJSDefinedTest.scala

```diff
@@ -565,6 +568,32 @@ class ScalaJSDefinedTest {
     assertEquals(100, dyn.foo())
   }
 
+  @Test def readonly_properties(): Unit = {
+    assumeFalse(
+        "Assuming strict mode semantics, which are not honored by Rhino",
+        Platform.executingInRhino)
+
+    // Named classes
+    @ScalaJSDefined
+    class Foo extends js.Object {
+      def bar: Int = 1
```

Comment:
We *could* make them immutable (using the `writable = false` flag to `defineProperty`), but we don't. Our "spec" doesn't mention making vals actually immutable on the js side. If we did, it could have bad performance impacts. We can reconsider this, but that would be a spec change.